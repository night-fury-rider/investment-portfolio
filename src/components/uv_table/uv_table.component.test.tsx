import { shallow } from 'enzyme';

import UVTable from './uv_table.component';


describe('Table Component Suite', ()=> {

  const tableHeaders = ['Test1', 'Test2'],
      tableRows = [
                  [1, 5],
                  [2, 7],
                  [3, 11]
                ];

  test('has correct snapshot', ()=> {
    expect(UVTable).toMatchSnapshot();
  });

  test('renders passed rows and headers', ()=> {
    const shallowComponent = shallow(<UVTable headers={tableHeaders} rows={tableRows}/>);
    expect(shallowComponent.text()).toContain(tableHeaders[0]);
    expect(shallowComponent.find('th').length).toEqual(tableHeaders.length);
    expect(shallowComponent.find('td').length).toEqual(tableHeaders.length * tableRows.length);
  });

  test('does not render table when no data is provided', ()=> {
    const shallowComponent = shallow(<UVTable headers={[]} rows={[[]]}/>);
    expect(shallowComponent.find('th').length).toEqual(0);
    expect(shallowComponent.find('td').length).toEqual(0);
  });
});
