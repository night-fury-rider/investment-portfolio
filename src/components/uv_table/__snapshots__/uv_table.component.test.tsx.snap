// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Suite has correct snapshot 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "compare": null,
  "type": [Function],
}
`;
